locale/ar/LC_MESSAGES/django.po

```diff
@@ -818,9 +818,9 @@ msgstr "بريد الالكتروني"
 #: templates/zerver/register.html:269
 #, python-format
 msgid ""
-"I agree to the <a href=\"%(root_domain_uri)s/terms\" target=\"_blank\" "
+"I agree to the <a href=\"%(root_domain_uri)s/policies/terms\" target=\"_blank\" "
 "rel=\"noopener noreferrer\">Terms of Service</a>."
-msgstr "أوافق على <a href=\"%(root_domain_uri)s/terms\" target=\"_blank\" rel=\"noopener noreferrer\">شروط الخدمة</a>"
+msgstr "أوافق على <a href=\"%(root_domain_uri)s/policies/terms\" target=\"_blank\" rel=\"noopener noreferrer\">شروط الخدمة</a>"
```

Comment:
Locale files should be left as-is; they'll get reverted when we next sync from Transifex anyway.

Reply:
That's unfortunate, since it means translators will have to re-translate the strings, unless I misunderstand how this works?

Reply:
That’s correct.

Ideally if we’re going to force translators to retranslate a string, we should do it in a way that abstracts away the HTML from translators so we don’t have to force even more retranslations in the future to change the HTML. That’s the point of the [`<z-foo>` mechanism](https://zulip.readthedocs.io/en/latest/translating/internationalization.html#frontend-translations) in the frontend JS and Handlebars. We can do it a bit more jankily here with `trans` bindings:

```jinja
{% macro begin_link() %}<a href="{{root_domain_uri}}/policies/terms">{% endmacro %}
{% macro end_link() %}</a>{% endmacro %}

{% trans begin_link=begin_link(), end_link=end_link() %}
I agree to the {{begin_link}}Terms of Service{{end_link}}
{% endtrans %}
```

Reply:
Oh, `{% set %}`…`{% endset %}` is a bit terser:

```jinja
{% set begin_link %}<a href="{{root_domain_uri}}/policies/terms">{% endset %}
{% set end_link %}</a>{% endset %}
{% trans %}I agree to the {{begin_link}}Terms of Service{{end_link}}{% endtrans %}
```

Reply:
Lint doesn't like that:
```
templates | Traceback (most recent call last):
templates |   File "/home/zulipdev/zulip/tools/check-templates", line 132, in <module>
templates |     check_our_files(args.modified, args.all_dups, args.fix, args.targets)
templates |   File "/home/zulipdev/zulip/tools/check-templates", line 44, in check_our_files
templates |     check_html_templates(by_lang["html"], all_dups, fix)
templates |   File "/home/zulipdev/zulip/tools/check-templates", line 104, in check_html_templates
templates |     tokens = validate(fn)
templates |   File "/home/zulipdev/zulip/tools/../tools/lib/template_parser.py", line 481, in validate
templates |     state.matcher(token)
templates |   File "/home/zulipdev/zulip/tools/../tools/lib/template_parser.py", line 439, in f
templates |     raise TemplateParserError(
templates | tools.lib.template_parser.TemplateParserError:
templates |                     fn: templates/zerver/accounts_accept_terms.html
templates |                    Mismatched tags: (a != set)
templates |                     start:
templates |                         <a href="{{ root_domain_uri }}/terms" target="_blank" rel="noopener noreferrer">
templates |                         line 52, col 45
templates |                     end tag:
templates |                         set
templates |                         line 52, col 125
templates |

Reply:
> That's unfortunate, since it means translators will have to re-translate the strings, unless I misunderstand how this works?

I think Transifex might show them the similar previous translation, such that it's not a huge pain, but yes ideally we'd use something like this `set` syntax for these. 

I think we can merge this after dropping these hunks and then consider getting `set` working in Django templates to be a separate sweep; there's like dozens of strings where we could be using it:
```
$ grep href locale/en/LC_MESSAGES/django.po  | wc
     39     220    2743
```